ftw/solr/indexer.py

```diff
@@ -18,6 +18,16 @@ class SolrIndexQueueProcessor(object):
 
     def index(self, obj, attributes=None):
         """Index the given object."""
+        if attributes:
+            attributes = tuple(attributes)
```

Comment:
Why are you casting to a tuple?

Reply:
Because `attributes` can be either a tuple or a list, depending on what was passed as `idxs` in a `reindexObject()` call.

The mechanism to extend the sequence therefore needs to work for immutable types like tuples as well (so, re-assignment instead of mutation). Because lists and tuples can't be concatenated though, we need to cast the sequence to a tuple anyway in the "add-path-depth" condition.

If we were to do it *only* in that condition though, the `attributes` that get passed along to the handler would be of a different type (list or tuple), depending on whether the condition was met. I didn't like that at all, and therefore made sure we alwas pass an immutable tuple to the handler. It also made for very confusing tests when the mock calls kept changing their argument types depending on how exactly the `index()` method was invoked.

Now that I moved the adding of `path_depth` to the handler's `get_data()` however, this has become unnecessary.